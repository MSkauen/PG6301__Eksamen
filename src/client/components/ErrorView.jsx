import React from "react";

export function ErrorView({ error, reload }) {
  if (error.status === 401) {
    return (
      <div>
        You are not logged in.
        <a href={"/api/login"} target={"_self"}>
          <button>Log in</button>
        </a>
      </div>
    );
  }
  return (
    <>
      <div>Something went wrong: {error.toString()}</div>
      {reload && <button onClick={reload}>Try Again</button>}
    </>
  );
}
