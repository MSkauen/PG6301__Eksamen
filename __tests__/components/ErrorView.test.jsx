import { ErrorView } from "../../src/client/components/ErrorView";
import TestRenderer from "react-test-renderer";

import React from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";

describe("error view", () => {
  it("show error view on dom", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(<ErrorView error={401} />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(container.querySelector("div").textContent).toEqual(
      "Something went wrong: 401"
    );
  });

  it("show error view on dom", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(<ErrorView error={404} />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(container.querySelector("div").textContent).toEqual(
      "Something went wrong: 404"
    );
  });

  it("show error view on dom", () => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      ReactDOM.render(<ErrorView error={500} />, container);
    });

    expect(container.innerHTML).toMatchSnapshot();
    expect(container.querySelector("div").textContent).toEqual(
      "Something went wrong: 500"
    );
  });
});
