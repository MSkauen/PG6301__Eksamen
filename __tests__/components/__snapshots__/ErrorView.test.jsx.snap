// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error view show error view on dom 1`] = `"<div>Something went wrong: 401</div>"`;

exports[`error view show error view on dom 2`] = `"<div>Something went wrong: 404</div>"`;

exports[`error view show error view on dom 3`] = `"<div>Something went wrong: 500</div>"`;
